pkg/ruler/compat.go

```diff
@@ -18,24 +19,44 @@ type Pusher interface {
 	Push(context.Context, *client.WriteRequest) (*client.WriteResponse, error)
 }
 type appender struct {
-	pusher  Pusher
-	labels  []labels.Labels
-	samples []client.Sample
-	userID  string
+	pusher        Pusher
+	labels        []labels.Labels
+	samples       []client.Sample
+	userID        string
+	refMap        map[uint64]labels.Labels
+	reverseRefMap map[uint64]uint64
+	lastRef       uint64
```

Comment:
Shouldn't these maps/lastRef being for TSDB and not appender? The TSDB contract is that the ref you receive from one appender can be used for another appender as well. I think we're breaking the contract here.

Reply:
That sounds right